src/harness/sas_objects.py

```diff
@@ -0,0 +1,387 @@
+        """
+        heartbeat_requests = []
+        for grant_object in self.grant_objects.values():
+            if(grant_object.isGrantActive()):
+                heartbeat_request = grant_object.constructHeartbeatRequest()
+                heartbeat_requests.append(heartbeat_request)
+        return heartbeat_requests
+
+    def getOperationParamsOfAllAuthorizedGrants(self):
+        """
+        Returns the list of operation params of all authorized grants
+        """
+        operation_params = []
+        for grant_object in self.grant_objects.values():
+            if(grant_object.isGrantAuthorizedInLastHeartbeat()):
+                operation_param = grant_object.getRequestOperationParam()
+                operation_params.append(operation_param)
+        return operation_params
+
+
+class DomainProxy():
+    """
+        Holds the domain proxy related parameters
+    """
+
+    def __init__(self, ssl_cert, ssl_key, testcase):
+        """
+        Args:
+            ssl_cert: Path to SSL cert file
+            ssl_key: Path to SSL key file
+            testcase: test case object from the caller
+        """
+        self.ssl_cert = ssl_cert
+        self.ssl_key = ssl_key
+        self.cbsd_objects = {}
+        self.testcase = testcase
+
+    def initialize(self, registration_requests, grant_requests):
+        """
+        Args
+            registration_requests:  A list of dictionary with a single key-value pair
+                where the key is"registrationRequest" and the value is a list of
+                individual CBSD registration  requests (each of which is itself a dictionary).
+            grant_requests: A list of dictionary with a single key-value pair
+                where the key is"grantRequest" and the value is a list of
+                individual CBSD grant requests (each of which is itself a dictionary).
+                There should be exactly one grant request per registration request.
+        """
+        #Checking if the number of registration requests matches number of grant requests
+        #There should be exactly one grant request per registration request.
+        assert (len(grant_requests) == len(registration_requests))
+        cbsd_ids = self.testcase.assertRegistered(registration_requests)
+
+        #Make one CBSD object per successful grant request
+        for index, grant_request in enumerate(grant_requests):
+            grant_request["grantRequest"][0]["cbsdId"] = cbsd_ids[index]
+            grant_response = self.testcase._sas.Grant(grant_request, self.ssl_cert, self.ssl_key)[
+                'grantResponse'][0]
+            if grant_response['response']['responseCode'] == 0:
+                cbsdobject = Cbsd(cbsd_ids[index], registration_requests[index],
+                                  [grant_response['grantId']], [grant_request])
+                self.cbsd_objects.update(
+                    {grant_response['cbsdId']: cbsdobject})
```

Comment:
It will probably be more readable if you use either `cbsd_id[index]` or `cbsd_object.cbsdId()`.

Reply:
Logic   has been changed